v2/tools/asoctl/internal/importing/arm_resource_importer.go

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package importing
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
+
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/reflecthelpers"
+	"github.com/Azure/azure-service-operator/v2/internal/version"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/pkg/names"
+
+	armruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/runtime"
+	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+// ARMResourceImporter is an importer for ARM based resources.
+// Create one using the factory method on ResourceImporter.
+type ARMResourceImporter struct {
+	ResourceImporter
+	client               *azruntime.Pipeline
```

Comment:
As mentioned below, I think you should be able to use `genericarmclient` as the client here, rather than `azruntime.Pipeline`, which I _think_ should make your life easier.

If you run into issues with `genericarmclient` that it's too focused on delivering for the controller we should modify it to work for this case too. I'm specifically thinking things like the fact it has a required `metrics *metrics.ARMClientMetrics` which probably doesn't make sense in this context. We could switch that to a `WithMetrics` optional method though possibly, and modify the client so that if it doesn't have metrics regsitered it doesn't panic.

Reply:
Done.